acquisitions/src/main/resources/thunderjet/mod-orders/features/piece-deletion-restriction.feature

```diff
@@ -0,0 +1,124 @@
+Feature: Piece deletion restrictions from order and order line
+
+  Background:
+    * url baseUrl
+    * callonce loginAdmin testAdmin
+    * def okapitokenAdmin = okapitoken
+    * callonce loginRegularUser testUser
+    * def okapitokenUser = okapitoken
+    * def headersAdmin = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenAdmin)', 'Accept': 'application/json' }
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenUser)', 'Accept': '*/*' }
+    * configure headers = headersUser
+
+    * callonce variables
+
+
+  Scenario: Avoid deltion of piece when order has status 'Closed' and poLine cost quantity '1' and ReceivingWorkflow 'Syncronized'
+    increase piece quantity and delete successfully
+    * def orderId = call uuid
+    * def poLineId = call uuid
+    * def newPieceId = call uuid
+    * def fundId = call uuid
+    * def budgetId = call uuid
+
+    # 0. Create finances
+    * configure headers = headersAdmin
+    * call createFund { 'id': '#(fundId)' }
+    * call createBudget { 'id': '#(budgetId)', 'allocated': 10000, 'fundId': '#(fundId)' }
+
+    # 1. Create order
+    * def v = call createOrder { 'id': '#(orderId)' }
+
+    # 2. Create order line with quantity 1 and Receiving Workflow 'Synchronized order and receipt quantity' = false
+    * def v = call createOrderLine { 'id': '#(poLineId)', 'orderId': '#(orderId)', 'quantity': 1, 'checkinItems': false }
+
+    # 3. Open Order
+    * def v = call openOrder { 'id': '#(orderId)' }
+
+    # 4. Get Piece by poLineId
+    Given path 'orders/pieces'
+    And param query = 'poLineId==' + poLineId
+    When method GET
+    Then status 200
+    And match $.totalRecords == 1
+    * def piece = $.pieces[0]
+    * def pieceId = piece.id
+
+    # 5. Verify validation for delete Piece with Order Status 'Open'
+    Given path 'orders/pieces', pieceId
+    When method DELETE
+    Then status 422
+    And match $.errors[0].message == "The piece cannot be deleted because it is the last piece for the poLine in with Receiving Workflow 'Synchronized order and receipt quantity' and cost quantity '1'"
```

Comment:
Can we reuse this message and define it once in Background?

Reply:
Thanks, moved to background